runtime/src/app/types.ts

```diff
@@ -8,21 +8,11 @@ type Child = {
 	component?: Component;
 };
 
-export interface ComponentConstructor {
-	new (options: { target: Node, props: any, hydrate: boolean }): Component;
-	preload: (props: { params: Params, query: Query }) => Promise<any>;
-}
-
 export interface Component {
 	$set: (data: any) => void;
```

Comment:
`$set` is for Props, so maybe you could change this to

```ts
export interface Component<Props = any> {
	$set: (data: Partial<Props>) => void;
	$destroy: () => void;
}
```

Reply:
I no longer have a `Props` type

Reply:
I meant a `Props` generic, because the `$set` method expects tlan object of input properties of the component, you might want to type this in a generic way.

Reply:
Ok. I think I will save tightening up the types for a follow up (which I think @ajbouh was interested in taking a stab at) and maybe you can lend a set of eyes to that. There's a whole bunch of places that would need to be changed if I'm going to make use of it in a meaningful way. Right now I'm mostly just trying to do the minimal to get the build configuration in and am afraid this PR has already grown pretty large

Reply:
Good call 👍